internal/services/postgres/postgresql_flexible_server_resource.go

```diff
@@ -400,8 +399,16 @@ func resourcePostgresqlFlexibleServer() *pluginsdk.Resource {
 			// storage_mb size...
 			storageTiers := storageTierMappings[newMb]
 
+			newTier = ""
```

Comment:
I assume that we don't need it anymore if using "IsExplicitlyNullInConfig" to fix the issue.